===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(swTimer_add:tnode)) at 37 (line 123, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/swoole-src/src/network/Timer.c
+++ b/table1/benchmarks/swoole-src/src/network/Timer.c
@@ -122,6 +122,7 @@ swTimer_node* swTimer_add(swTimer *timer, int _msec, int interval, void *data)
     int64_t now_msec = swTimer_get_relative_msec();
     if (now_msec < 0)
     {
+        free(tnode);
         return NULL;
     }
 
